code/Controllers/CMSSiteTreeFilter.php

```diff
@@ -203,13 +204,19 @@ protected function applyDefaultFilters($query)
             switch ($name) {
                 case 'Term':
                     $query = $query->filterAny(array(
-                        'URLSegment:PartialMatch' => $val,
+                        'URLSegment:PartialMatch' => Convert::raw2url($val),
```

Comment:
Perhaps we should use `rawurlencode()` as that's what is used here: https://github.com/silverstripe/silverstripe-cms/blob/4.6.0/code/Model/SiteTree.php#L425

Reply:
I tried your suggestion but this breaks the newly added unit test which indicates that this doesn't cover all cases.

Reply:
Interesting - that implies `SiteTree::get_by_link()` will be broken in some instances too?

Reply:
OK - I think that `rawurlencode()` is used in `get_by_link` because the assumption is that the provided url would have all the conversions done already (they'd be in the actual URL that a request has been made to). If `URLSegmentFilter` was used, for example, then a search for `you&me` would return a page `you-and-me` even though you wouldn't expect that to be the case.